include/zephyr/drivers/bluetooth.h

```diff
@@ -0,0 +1,257 @@
+ *
+ * @param dev HCI device
+ * @param buf Buffer containing data to be sent to the controller.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_send(const struct device *dev, struct net_buf *buf)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	return api->send(dev, buf);
+}
+
+#if defined(CONFIG_BT_HCI_SETUP) || defined(__DOXYGEN__)
+/**
+ * @brief HCI vendor-specific setup
+ *
+ * Executes vendor-specific commands sequence to initialize
+ * BT Controller before BT Host executes Reset sequence. This is normally
+ * called directly after bt_hci_open().
+ *
+ * @note @kconfig{CONFIG_BT_HCI_SETUP} must be selected for this
+ * field to be available.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_setup(const struct device *dev, struct bt_hci_setup_params *params)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	if (api->setup == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->setup(dev, params);
+}
+#endif
+
+/**
+ * @}
+ */
+
+/* The following functions are not strictly part of the HCI driver API, in that
+ * they do not take as input a struct device which implements the HCI driver API.
+ */
+
+/**
+ * @brief Setup the HCI transport, which usually means to reset the
+ * Bluetooth IC.
+ *
+ * @note A weak version of this function is included in the H4 driver, so
+ *       defining it is optional per board.
+ *
+ * @param dev The device structure for the bus connecting to the IC
+ *
+ * @return 0 on success, negative error value on failure
+ */
+int bt_hci_transport_setup(const struct device *dev);
+
+/**
+ * @brief Teardown the HCI transport.
+ *
+ * @note A weak version of this function is included in the IPC driver, so
+ *		defining it is optional. NRF5340 includes support to put network core
+ *		in reset state.
+ *
+ * @param dev The device structure for the bus connecting to the IC
+ *
+ * @return 0 on success, negative error value on faulure
+ */
+int bt_hci_transport_teardown(const struct device *dev);
+
+/** Allocate an HCI event buffer.
+ *
+ * This function allocates a new buffer for an HCI event. It is given the
+ * event code and the total length of the parameters. Upon successful return
+ * the buffer is ready to have the parameters encoded into it.
+ *
+ * @param evt        Event OpCode.
+ * @param len        Length of event parameters.
+ *
+ * @return Newly allocated buffer.
+ */
+struct net_buf *bt_hci_evt_create(uint8_t evt, uint8_t len);
+
+/** Allocate an HCI Command Complete event buffer.
+ *
+ * This function allocates a new buffer for HCI Command Complete event.
+ * It is given the OpCode (encoded e.g. using the BT_OP macro) and the total
+ * length of the parameters. Upon successful return the buffer is ready to have
+ * the parameters encoded into it.
+ *
+ * @param op         Command OpCode.
```

Comment:
Ditto with a reference to the possible opcode values

Reply:
I added "HCI", in case that makes it any clearer. Btw, there's are exact copies of what's in `include/zephyr/drivers/bluetooth/hci_driver.h`. They're not really part of the HCI driver API, but I had to drag them along to not create scope creep in this PR (we really need to rethink and redesign this aspect of the HCI driver API).